### PR TITLE
Add v2tofhir accelerator

### PR DESCRIPTION
**Pull request by ThishaniLucas**

### CODE REVIEW

File: transformations/hl7v2-to-fhir/service.bal [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+service / on new http:Listener(9090) {
+
+    resource function post transform(http:Request request) returns http:Response|http:ClientError {
+        string textPayload = check request.getTextPayload();
+        json|error v2tofhirResult = v2tofhirr4:v2ToFhir(textPayload);
+        http:Response response = new;
+        if v2tofhirResult is json {
+            response.setJsonPayload(v2tofhirResult);
+            error? contentType = response.setContentType("application/fhir+json");
```
Comment (nirmal070125) [inline transformations/hl7v2-to-fhir/service.bal:28]: take from fhir r4 constant.

File: transformations/hl7v2-to-fhir/service.bal [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            if contentType is error {
+                response.statusCode = http:STATUS_INTERNAL_SERVER_ERROR;
+            }
```
Comment (nirmal070125) [inline transformations/hl7v2-to-fhir/service.bal:31]: please set a message as well.

**Review by nirmal070125 — APPROVED** [review]

File: transformations/hl7v2-to-fhir/service.bal (hunk 1 above)
Comment (ThishaniLucas) [reply transformations/hl7v2-to-fhir/service.bal]: done

File: transformations/hl7v2-to-fhir/service.bal (hunk 2 above)
Comment (ThishaniLucas) [reply transformations/hl7v2-to-fhir/service.bal]: done

File: transformations/hl7v2-to-fhir/service.bal [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    resource function post transform(http:Request request) returns http:Response|http:ClientError|error {
+        string textPayload = check request.getTextPayload();
+        json|error v2tofhirResult = v2tofhirr4:v2ToFhir(textPayload);
+        http:Response response = new;
+        if v2tofhirResult is json {
+            response.setJsonPayload(v2tofhirResult);
+            response.statusCode = http:STATUS_OK;
+            _ = check response.setContentType(r4:FHIR_MIME_TYPE_JSON);
+        } else {
+            response.setPayload("Unable to Convert HL7 Message to FHIR");
```
Comment (nirmal070125) [inline transformations/hl7v2-to-fhir/service.bal:32]: ```string `Unable to convert HL7 message to FHIR. Error:  ${v2tofhirResult.message()}.` ```

Comment (ThishaniLucas) [reply transformations/hl7v2-to-fhir/service.bal:32]: v2tofhirResult.message() always prints 'HL7ParserError'. The actual messages can be captured from v2tofhirResult.detail but they are too internal. That's why I thought of sending a generic message.

Comment (nirmal070125) [reply transformations/hl7v2-to-fhir/service.bal:32]: Use `v2tofhirResult.toBalString()`.

**Review by sameeragunarathne — APPROVED** [review]

**Pull request merged**